### PR TITLE
acl: JWT auth CLI

### PR DESCRIPTION
**Pull request by pkazmierczak**
This PR introduces CLI command adjustments for JWT-based auth.

### CODE REVIEW

File: command/acl_auth_method_create.go [hunk 1]
```diff
@@ -50,8 +50,8 @@ ACL Auth Method Create Options:
     between 1-128 characters and is a required parameter.
 
   -type
-    Sets the type of the auth method. Currently the only supported type is
-    'OIDC'.
+    Sets the type of the auth method. Currently the only supported types are
+    'OIDC' and 'JWT'.
```
Comment (lgfa29) [inline command/acl_auth_method_create.go:54]:
```suggestion
    Sets the type of the auth method. Supported types are 'OIDC' and 'JWT'.
```

Same for `update`. We also need to update the [website](https://github.com/hashicorp/nomad/blob/main/website/content/docs/commands/acl/auth-method/create.mdx?plain=1#L32-L33).

File: command/login.go [hunk 2]
```diff
@@ -50,7 +51,12 @@ Login Options:
 
   -oidc-callback-addr
     The address to use for the local OIDC callback server. This should be given
-    in the form of <IP>:<PORT> and defaults to "localhost:4649".
+    in the form of <IP>:<PORT> and defaults to "localhost:4649". It is only
+    required if using OIDC auth method type.
```
Comment (lgfa29) [inline command/login.go:55]:
```suggestion
    in the form of <IP>:<PORT> and defaults to "localhost:4649".
```

I think the flag name make this clear already?

File: command/login.go [hunk 3]
```diff
@@ -50,7 +51,12 @@ Login Options:
 
   -oidc-callback-addr
     The address to use for the local OIDC callback server. This should be given
-    in the form of <IP>:<PORT> and defaults to "localhost:4649".
+    in the form of <IP>:<PORT> and defaults to "localhost:4649". It is only
+    required if using OIDC auth method type. 
+
+  -login-token
```
Comment (lgfa29) [inline command/login.go:57]:
```suggestion
  -login-token=<token>
```

We usually document flags that take values to differentiate them from boolean types (like `-json` below), but I don't know how consistent we are about it 😅

File: command/login.go [hunk 4]
```diff
@@ -71,6 +77,7 @@ func (l *LoginCommand) AutocompleteFlags() complete.Flags {
 		complete.Flags{
 			"-method":             complete.PredictAnything,
 			"-oidc-callback-addr": complete.PredictAnything,
+			"-login-token":        complete.PredictNothing,
```
Comment (lgfa29) [inline command/login.go:80]:
```suggestion
			"-login-token":        complete.PredictAnything,
```

It doesn't _really_ matter in this case, but `PredictAnything` calls the predictor function, while `PredictNothing` doesn't.

**Review by lgfa29 — APPROVED** [review]
Minor nit-picking, but LGTM!

File: command/login.go (hunk 3 above)
Comment (pkazmierczak) [reply command/login.go:57]: eh we don't do this for any of the auth method commands...

**Pull request merged**